minetest/minetest
# Add register dialog to separate login/register

Fixes #8138

New users find Minetest's account system confusing. This change moves username/password to a new dialog, with login and register buttons added to the Join Game tab.

The old registration confirmation dialog is removed in favour of the new dialog. Instead of disabling register confirmation, users can completely disable split login/register using `enable_split_login_register = false`


![image](https://user-images.githubusercontent.com/2122943/167882843-4af4d0f1-e575-4013-801a-5dc12fc1ef36.png)
![image](https://user-images.githubusercontent.com/2122943/165139241-89af6ea4-735e-4970-a523-41c25d9e619d.png)
![image](https://user-images.githubusercontent.com/2122943/165139260-e6e1cdce-121b-4565-aea3-7e3b16a8de6a.png)
![image](https://user-images.githubusercontent.com/2122943/165139687-56b6ebe8-4ec7-4b66-a073-4ce9acbadab0.png)


## To do

- [x] Move delete favourite button to a better location
- [x] Error messages on register from login, and login from register
- [x] What to do when connecting to localhost?

## How to test

* Register:
    * Clear minetest.conf. Click register, confirm name field is empty and focused.
    * Fill in information, connect. Confirm server is marked as favourite.
    * Attempt to register on another server, confirm that name is remembered and password is focused
    * Test behaviour when passwords do not match
* Login:
    * Test correct log-in
    * Test attempt to register
    * Test incorrect password
* Test `enable_split_login_register = false` behaves as master, but without the register confirmation dialog
